builtin/providers/aws/structure.go

```diff
@@ -813,11 +813,20 @@ func flattenStepAdjustments(adjustments []*autoscaling.StepAdjustment) []map[str
 	return result
 }
 
-func flattenResourceRecords(recs []*route53.ResourceRecord) []string {
+func flattenResourceRecords(recs []*route53.ResourceRecord, typeStr string) []string {
 	strs := make([]string, 0, len(recs))
 	for _, r := range recs {
 		if r.Value != nil {
-			s := strings.Replace(*r.Value, "\"", "", 2)
+			s := *r.Value
+			if typeStr == "TXT" || typeStr == "SPF" {
+				// On Route53, these records are usually quoted.  This plugin defines the
+				// canonical format to be unquoted.  So when converting from the Route53 value
+				// to the Terraform config value, remove quotes.
+				l := len(s)
+				if l > 1 && s[0] == '"' && s[l-1] == '"' {
+					s = s[1:(l - 1)]
+				}
```

Comment:
When inputs are well-formed, I think `strings.Trim` would be equivalent.  Sadly, the plugin currently doesn't strictly validate inputs, so I'm worried about the behavior on strange inputs (e.g. `abc "def"` or `"abc" ""`).

My thinking was that the current code is slightly more predictable: if there are quotes at both ends, remove them, otherwise do nothing.

That said, maybe we should modify this plugin to be more strict about inputs?
- When pulling from Route53, remove all quotes and create a single string.
- When pushing to Route53, split on the maximum length and re-quote every segment.

Two downsides:
- I *think* this might have to be a backwards-incompatible change.  But maybe it's worth it?
- Values won't round-trip verbatim, but they'll be semantically equivalent.